api/db/populate-friendships.js

```diff
@@ -0,0 +1,85 @@
+const { getRandomSubset, getRandomDate, createPrompt } = require('./utils');
+
+const User = require('../models/User');
+const Friendship = require('../models/Friendship');
+
+let displayLogs;
+
+const createFriendship = async (friendshipParams) => {
+  const friendship = new Friendship({
+    requestant: friendshipParams.requestant,
+    receivant: friendshipParams.receivant,
+    accepted: friendshipParams.accepted,
+    acceptedAt: friendshipParams.acceptedAt,
+    lastMessageAt: friendshipParams.lastMessageAt,
+  });
+  await friendship.save();
+
+  if (displayLogs) {
+    console.log('\n', '\x1b[0m', `New friendship created: ${friendship}`);
+  }
+  return friendship;
+};
+
+const createFriendships = async (verbose) => {
+  console.log('\n', '\x1b[0m', 'Populating friendships collection...');
+  displayLogs = verbose;
+
+  let usersList = await User.distinct('_id');
+  const friendshipsArray = await Friendship.find();
+  let recordCount = friendshipsArray.length;
+
+  let proceed = true;
+  if (recordCount > 0) {
+    proceed = await createPrompt(
+      'Some users already have friends. Would you like to add friends to users that already contain friends?'
+    );
+  }
+
+  for (const user of usersList) {
+    const containFriends = friendshipsArray.some(
+      (friendship) =>
+        friendship.requestant.toString() === user.toString() ||
+        friendship.receivant.toString() === user.toString()
+    );
+    if (!proceed && containFriends) continue;
+
+    // split current user from all others
+    const tmpList = usersList.filter((userValue) => userValue != user);
```

Comment:
Why not use the `toString` method here too and use `!==` instead of `!=`?

Reply:
Sorry, I come from a Java and Ruby on Rails background and often type just the two symbols when evaluating. It's a typo. I'll change it, but we shouldn't need the `toString()` because we are evaluating a user object with another user object and not just their `id`'s.